### ISSUE TITLE
Add rlpHeader param to SignatureVerification structure

### ISSUE THREAD

**Issue by P1sar** [ChainSafe/chainbridge-celo#118]
According to newer investigation on Celo and our sides, was defined that to make Proof valid we should somehow prove that TxRootHash belongs to BlockHash that was signed by provided Aggregated Public Key of block validators ( here and next APK). To prove this wee need to add additional param
## Implementation details

Add rlpHeader - rlp encoded of concatenated block header data. This RLP encoding usually used to calculate BlockHash. 
SO basically you need to make rlp.Encode over header struct. (https://github.com/celo-org/celo-blockchain/blob/master/core/types/block.go#L70)
## Testing details

the tests is passing
## Acceptance Criteria

the tests is passing
SignatureVerification contains RLPHeader
